src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,133 @@
+                    ui.printMessage(pastTasks.toString());
+                } else if (parser.getType(command).equals("find")) {
+                    TaskList results = new TaskList();
+                    for (Task t: pastTasks.taskList) {
+                        String taskToFind = Parser.getTaskToFindName(command);
+                        if (t.name.contains(taskToFind)) {
+                            results.add(t);
+                        }
+                    }
+
+                    ui.printMessage(results.toString());
+                } else if (parser.getType(command).equals("unmark")) {
+                    int index = Integer.parseInt(command.split(" ")[1]) - 1;
+                    pastTasks.get(index).isDone = false;
+
+                    ui.printMessage("OK, I've marked this task as not done yet:\n  "
+                            + pastTasks.get(index)
+                    );
+                } else if (parser.getType(command).equals("todo")) {
+                    if (command.length() == 4) {
+                        throw new InputMismatchException("The description of a todo cannot be empty.");
+                    }
+
+
+                    pastTasks.add(parser.parseTodos(command));
+                    storage.save();
+
+                    pastTasks.add(parser.parseTodos(command));
+                    storage.save();
+
+                    ui.printMessage(String.format("""
+                            Got it. I've added this task:
+                            %s
+                            Now you have %d %s in the list.""",
+                            pastTasks.get(pastTasks.size() - 1),
+                            pastTasks.size(),
+                            pastTasks.size() == 1 ? "task" : "tasks"
+                    ));
+                } else if (parser.getType(command).equals("deadline")) {
+                    pastTasks.add(parser.parseDeadlines(command));
+                    storage.save();
+
+                    ui.printMessage(String.format("""
+                            Got it. I've added this task:
+                            %s
+                            Now you have %d %s in the list.""",
+                            pastTasks.get(pastTasks.size() - 1),
+                            pastTasks.size(),
+                            pastTasks.size() == 1 ? "task" : "tasks"
+                    ));
+                } else if (parser.getType(command).equals("event")) {
+                    pastTasks.add(parser.parseEvent(command));
+                    storage.save();
+
+                    ui.printMessage(String.format("""
+                            Got it. I've added this task:
+                            %s
+                            Now you have %d %s in the list.""",
+                            pastTasks.get(pastTasks.size() - 1),
+                            pastTasks.size(),
+                            pastTasks.size() == 1 ? "task" : "tasks"
```

Comment:
Perhaps this string could be abstracted out into a method if you are using it in the same format for the 3 different commands